packages/foam-vscode/src/services/provider.ts

```diff
@@ -0,0 +1,6 @@
+import { Resource, ResourceProvider } from 'foam-core';
+
+export interface FoamResourceProvider extends ResourceProvider {
+  getNoteTooltip(resource: Resource): string;
```

Comment:
What's the thinking behind having these display-specific methods on the resource provider as opposed to just utility functions? 

Is the FoamResourceProvider going to have multiple different implementations that need different implementations of these methods?